apps/web/src/hooks/useFileOperations.ts

```
@@ -151,3 +151,102 @@ export function useUploadFile() {
+			const contentDisposition = response.headers.get("Content-Disposition");
+			let filename = fileName || "download";
+			if (contentDisposition) {
+				const filenameMatch = contentDisposition.match(/filename="(.+)"/);
+				if (filenameMatch && filenameMatch[1]) {
+					filename = filenameMatch[1];
+				}
+			}
+
+			// Track download progress if the response supports it
+			const contentLength = response.headers.get("Content-Length");
+			const total = contentLength ? parseInt(contentLength, 10) : 0;
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)